### PR TITLE
adding certificate updates info

### PR DESCRIPTION
**Pull request by wduraes**
Adding information regarding recent update plans for Baltimore certificates. Since C SDK has this cert on the samples, we need to add instructions on how to deal with it.

### CODE REVIEW

File: certs/readme.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+For sample purposes, C SDK samples leverage certificates on the `certs.c` file. When creating solutions based on the C SDK you should consider :
+
+1. For production environments, the IoT Hub certificate must be validated during the TLS handshake.
+2. We use the Baltimore root CA to validate IoT Hub server certificate (for the regions that use this certificate)
+3. For other regions (and private cloud environments) the appropriate root CA shall be used
+
+## Certificate update strategy
+
+As shared on [this blog post](https://techcommunity.microsoft.com/t5/internet-of-things/azure-iot-tls-changes-are-coming-and-why-you-should-care/ba-p/1658456), Microsoft began phasing out her older PKI provided by Baltimore Cybertrust Root this August in favor of new PKI providers. For Azure IoT, this means that Azure IoT services will serve TLS certificates that will be rooted to the following:
```
Comment (ewertons) [inline certs/readme.md:11]: Microsoft began phasing out **its**

File: certs/readme.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+- [Baltimore CyberTrust Root](https://cacerts.digicert.com/BaltimoreCyberTrustRoot.crt): Although this root is expiring in 2025, IoT Hub and Device Provisioning Service (DPS) will remain chained to this root until 2023 and a plan with defined timelines will be announced in March, 2021.
```
Comment (ewertons) [inline certs/readme.md:13]: announced on March, 2021.

File: certs/readme.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+## Certificate update strategy
+
+As shared on [this blog post](https://techcommunity.microsoft.com/t5/internet-of-things/azure-iot-tls-changes-are-coming-and-why-you-should-care/ba-p/1658456), Microsoft began phasing out her older PKI provided by Baltimore Cybertrust Root this August in favor of new PKI providers. For Azure IoT, this means that Azure IoT services will serve TLS certificates that will be rooted to the following:
+
+- [Baltimore CyberTrust Root](https://cacerts.digicert.com/BaltimoreCyberTrustRoot.crt): Although this root is expiring in 2025, IoT Hub and Device Provisioning Service (DPS) will remain chained to this root until 2023 and a plan with defined timelines will be announced in March, 2021. 
+
+- [Digicert Global Root G2](https://cacerts.digicert.com/DigiCertGlobalRootG2.crt): This is the new root that most of public Azure is chained to including all Azure IoT Services except IoT Hub and DPS as per 1
+
+- MS-PKI: Microsoft is eventually going to move public Azure to her own public PKI that has the following two roots:
```
Comment (ewertons) [inline certs/readme.md:17]: public Azure to **its** own

File: certs/readme.md [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+  - [RSA](http://www.microsoft.com/pkiops/certs/Microsoft%20RSA%20Root%20Certificate%20Authority%202017.crt) - Microsoft RSA Root Certificate Authority 2017
+  - [ECC](https://www.microsoft.com/pkiops/certs/Microsoft%20EV%20ECC%20Root%20Certificate%20Authority%202017.crt) - Microsoft EV ECC Root Certificate Authority 2017
+
+In order for devices to remain connected to Azure, Microsoft recommends adding all these roots to your device/certificate store. If you have a memory constrained device then Baltimore root will be the certificate to use for TLS until 2023, after which the guidance that will be provided then can be used.
```
Comment (ewertons) [inline certs/readme.md:22]: I am not sure this makes much sense to be included:

"If you have a memory constrained device, Baltimore root will be the certificate to use for TLS until 2023, after which the guidance that will be provided then can be used."

Seems redundant.

**Review by ewertons — CHANGES_REQUESTED** [review]

**Review by ericwol-msft — APPROVED** [review]

File: certs/readme.md (hunk 1 above)
Comment (CIPop) [reply certs/readme.md:11]: I'm a bit confused by the statement: the referenced blog is saying that no action is required and that some of the intermediate certificate authorities are changing between Aug and Nov 2020.

Comment (wduraes) [reply certs/readme.md]: we removed the redundant information from this file and added a link to the official source of information: the blog post.

File: certs/readme.md (hunk 3 above)
Comment (wduraes) [reply certs/readme.md]: we removed the redundant information from this file and added a link to the official source of information: the blog post.

File: certs/readme.md (hunk 4 above)
Comment (wduraes) [reply certs/readme.md]: we removed the redundant information from this file and added a link to the official source of information: the blog post.

File: certs/readme.md (hunk 2 above)
Comment (wduraes) [reply certs/readme.md]: we removed the redundant information from this file and added a link to the official source of information: the blog post.

**Review by ewertons — APPROVED** [review]
Suggestion:

### Certificates - Important to know

The Azure IoT Hub certificates presented during TLS negotiation shall be always validated using the appropriate root CA certificate(s).
The samples in this repository leverage the certificates in [`certs.c`](https://github.com/Azure/azure-iot-sdk-c/blob/master/certs/certs.c) for the United States, Blackforest and Mooncake regions.
For other regions (and private cloud environments), please use the appropriate root CA certificate.

**Comment by CIPop** [comment]
@ewertons I would only add to this:

> The samples in this repository leverage the certificates in certs.c 

Always prefer using the local system's Trusted Root Certificate Authority store instead of hardcoding the certificates (i.e. using `certs.c` such as our samples require in certain combinations). A couple of examples:

- For Windows, Schannel will automatically pick up CA certificates from the store managed using `certmgr.msc` .
- For Debian Linux, OpenSSL will automatically pick up CA certificates from the store installed using `apt install ca-certificates`. Adding a certificate to the store is described here: http://manpages.ubuntu.com/manpages/precise/man8/update-ca-certificates.8.html

**Pull request merged**